cmd/deploy.go

```diff
@@ -390,7 +390,7 @@ func (dc *deployCmd) run() error {
 		os.Exit(1)
 	}
 
-	template, parameters, err := templateGenerator.GenerateTemplate(dc.containerService, engine.DefaultGeneratorCode, BuildTag)
+	template, parameters, err := templateGenerator.GenerateTemplateV2(dc.containerService, engine.DefaultGeneratorCode, BuildTag)
```

Comment:
Let's merge this functional changeset without actually flipping the switch. This allows us to get this large delta into master quickly, to reduce the overhead of a long-lived branch that touches so much functional area.

Reply:
Sure. If there are no other review comments, I can revert this change. Otherwise, I'd need this for the e2e validations to actually happen.

Reply:
Also, because I'm adding a lot of review suggestions, keep this in until after we're ready to merge, so that we continue to conveniently test the changes. :)